source/agora/consensus/EnrollmentManager.d

```diff
@@ -587,4 +672,13 @@ unittest
     man.getUnregisteredEnrollments(enrolls);
     assert(enrolls.length == 3);
     assert(enrolls.isStrictlyMonotonic!("a.utxo_key < b.utxo_key"));
+
+    // test serialization/deserializetion for pre-images
+    auto preimages_1 = man.getPreimages();
+    auto preimages_2 = man.loadPreimages();
+    int true_count;
+    foreach (i; 0 .. 1008)
+        if (preimages_1[i] == preimages_2[i])
+            true_count++;
+    assert(true_count == 1008);
```

Comment:
There is a missing check that the preimages are actually preimages. You can use this:

```D
import std.algorithm;
assert(equal!((a, b) => a.hashFull() == b)(preimages_1[0 .. $ - 1], preimages_1[1 .. $]));
```